### PR TITLE
Improve Profile.Icon UI in settings

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Improves the UI for the Profile.Icon setting by adding an "Icon Type" combo box. This allows the user to pick from multiple options:
- None: sets the icon to "none" which is interpreted as no icon
- Built-in Icon: presents a combo box that enumerates the Segoe MDL 2 assets
- Emoji: presents a text box with a hint to open the emoji picker
- File: presents a text box to input the path of the image to use

Additionally, the rendered icon is displayed in the setting container. If "none", "none" is presented to the user (localized).
✅ Verified as accessible using Accessibility Insights
#10000

### CODE REVIEW

**Comment by carlos-zamora** [comment]
## Demo
{0912FB46-81D8-4804-A76E-3BDBA111389A}

{6106B4C6-3248-4FFA-8D84-8F6942AC182D}

{5A94FC34-28CF-4A02-A754-61C4937799C1}


{0DD7BEBE-03CB-4F18-88CC-7C4BBB210CD0}

**Comment by DHowett** [comment]
FWIW should we rename "hidden" to the more obvious "no icon"? Hidden suggests that there _is an icon_, but that it is hidden.

**Comment by carlos-zamora** [comment]
Feedback from team sync (10/7):
- Drag and drop would be cool, but there isn't really a nice spot for it. (maybe drop on the "+ add sound" button)
- (nit) take advantage of more width?
- [test]: sound is playing, but we navigate to another page
- take the users home directory and replace it with ~
   - [idea] display the filename, and display the directory path separately as a subtitle
      - [idea] only display the directory if there's multiple files with the same name
- [idea] when sound is added, we copy it over to the local folder
   - punted at this point
- We should validate that the files exist
   - maybe strikethrough the item that didn't load?
   - definitely don't try and play sound!
- Bell style interaction
   - have a preview
   - add a warning if "style: audible" is off
   - In "Bell sound" help text, say "you will need to enable "Bell notification style, audible"
   - in the future, combine bell style and sound
- [bug] if you close the file picker, we add an empty entry

**Comment by carlos-zamora** [comment]
## Demo

Built-In
Emoji
File
None

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp [hunk 1]
```diff
@@ -98,6 +127,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _defaultAppearanceViewModel.IsDefault(true);
     }
 
+    void ProfileViewModel::_UpdateBuiltInIcons()
+    {
+        _BuiltInIcons = single_threaded_vector<IInspectable>();
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:132]: fwiw right now we do this once per profile when we could do it once per settings UI

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp [hunk 2]
```diff
@@ -328,24 +409,89 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
...
+            // Setting the member here means that we early exit at the beginning of the function
+            //  because _currentIconType == value.
+            _currentIconType = value;
+
+            // Switched to...
+            switch (unbox_value<IconType>(value.as<Editor::EnumEntry>().EnumValue()))
+            {
+            case IconType::None:
+            {
+                Icon(HideIconValue);
```
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:455]: do we need to notify changes here?

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:455]: or, do we only notify below (R456) because we want the text box to update

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp (hunk 1 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:132]: Nah, `_BuiltInIcons` is static so it's a shared instance across all ProfileVMs.

And we only call it twice:
https://github.com/microsoft/terminal/blob/bb328969730ed5a04db24f07e504fe1584d962e0/src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp#L116-L119

https://github.com/microsoft/terminal/blob/bb328969730ed5a04db24f07e504fe1584d962e0/src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp#L165-L168

Both of which are wrapped in a null-check.

Confirmed w/ breakpoint in `_UpdateBuiltInIcons()` and poking around SUI (and closing/reopening SUI).

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp (hunk 2 above)
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:455]: Oh! My bad. The `_NotifyChanges(L"Icon");` on R456 is redundant. `Icon()` automatically notifies changes. Removing R456. Must've been left over from an earlier iteration 😅

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp (hunk 1 above)
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:132]: Oh jeez you're a step ahead of me!

File: src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp [hunk 3]
```diff
@@ -98,6 +127,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _defaultAppearanceViewModel.IsDefault(true);
     }
 
+    void ProfileViewModel::_UpdateBuiltInIcons()
+    {
+        _BuiltInIcons = single_threaded_vector<IInspectable>();
+        for (auto& [val, name] : s_SegoeFluentIcons)
+        {
+            _BuiltInIcons.Append(make<EnumEntry>(hstring{ name }, box_value(val)));
+        }
```
Comment (lhecker) [inline src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:136]: I'd recommend the "first construct a `std::vector`, then move it into an `IVector`" approach here given that you append a lot of items.

BTW it's a funny thought but technically it's completely 100% possible to create a custom HSTRING that does not the need a heap allocation. The HSTRING struct is part of the ABI.

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp:136]: https://learn.microsoft.com/en-us/windows/win32/api/winstring/nf-winstring-windowscreatestringreference eh

**Review by zadjii-msft — APPROVED** [review]

**Pull request merged**